aten/src/ATen/native/native_functions.yaml

```diff
@@ -14720,6 +14720,13 @@
   dispatch:
     CompositeExplicitAutograd: wait_tensor
 
+- func: all_reduce_coalesced(Tensor[] self, str reduceOp, str tag, int[] ranks, int group_size) -> Tensor[]
```

Comment:
Why do you make this a native function? You don't actually have any implementation for it here.
You can DEF new implementations with [torch.library](https://pytorch.org/docs/stable/library.html?highlight=torch+library#torch.library.Library). You can then do the exact same thing as you do here for it (even easily add it to your distributed namespace). And adding it to library makes it an op so it can live in the inductor graph the exact same way.

Reply:
IIRC, we ran into issues when defining it using `torch.library`.
I'll have to try it in a separate PR to refresh my memory.

Reply:
Another reason is that we want to move the implementation to C++ eventually as we need to support c++ backends

Reply:
You can still register the impl from c++ for a function that was defined in python :D

Reply:
You mean in an export context? Why not? You can still provide a C++ implementation to your custom no python runner.

Reply:
i think the reason we did this was to have a wya to get codegen for backwards from derivatives.yaml.

since that time we (gave up? postponed?) implementing backwards.  If we don't need codegen for new plan for backwards, i'm good with moving out of native and into py.

cc @kumpera

Reply:
Alban helped me figure out how to get dispatch to work as we need from python.

I guess we down to figuring out what to do with torch::deploy before moving on with this - instead of going the opposite direction of moving the ops implementations to c++